### ISSUE TITLE
Unable to nullify sub chart's values without explicitly specifying parents values file

### ISSUE THREAD

**Issue by abhinaybyrisetty** [helm/helm#10812]
Output of helm version:
version.BuildInfo{Version:"v3.7.2", GitCommit:"663a896f4a815053445eec4153677ddc24a0a361", GitTreeState:"clean", GoVersion:"go1.16.10"}

Output of kubectl version:
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T16:31:21Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.10", GitCommit:"8152330a2b6ca3621196e62966ef761b8f5a61bb", GitTreeState:"clean", BuildDate:"2021-08-11T18:00:37Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): Any cloud

Description:
I have been trying to nullify values in sub chart using parent chart's values.yaml.

**values.yaml of parent chart**

subchartx:
  section:
    key: 

In values.yaml of subchartx, a default value is assigned for key as below

section:
  key: mydefaultvalue
  
When I install the parent chart without specifying any values.yaml file, I expect it to take parent's values.yaml as the default values file.
However, that specific key is being nullified only if --values values.yaml is used in the installation command.

If I don't explicitly pass --values values.yaml while installing/upgrading, it takes the default value defined in the values.yaml of subchartx

Please let me know if you need any additional information.

**Comment by bacongobbler** [comment]
Yep, looks like a duplicate. Going to close this in favour of #10795 so we can track the discussion in one ticket. Thanks.
